internal/controller/server.go

```diff
@@ -332,22 +352,36 @@ func (s *MyServer) actionBatch(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *MyServer) actionAPIUrls(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(r.Context(), maxDBExecuteTime)
+
+	defer cancel()
+
 	userid, err := s.getUser(r)
 
+	if err != nil {
+		userid, err = s.sendAuthToken(w)
+
+		if err != nil {
+			s.actionError(w, "AUTH NEED BUT CAN'T:"+fmt.Sprintf("%s", err))
```

Comment:
```suggestion
			s.actionError(w, "AUTH NEED BUT CAN'T:"+err.Error())
```
Функции fmt.* с подстановкой очень прожорливы. Так что там, где можно применить стандартное преобразование в строку, лучше ограничиться этим преобразованием и конкатенацией.

Reply:
Поправил.